wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationService.swift

```diff
@@ -283,6 +285,54 @@ public final class ConversationService: ConversationServiceInterface {
         }
     }
 
+    private func handleMLSConversationIfNeeded(for conversationObjectId: NSManagedObjectID, participantIds: Set<QualifiedID>) async throws {
+        guard let syncContext = await context.perform({ self.context.zm_sync }) else {
+            assertionFailure("handleMLSConversationIfNeeded must be done on syncContext")
+            return
+        }
+        guard let syncConversation = await syncContext.perform({
+            let conversation = try? syncContext.existingObject(with: conversationObjectId) as? ZMConversation
+
+            guard conversation?.messageProtocol == .mls else {
+                // proteus: nothing to do for proteus, see action handler
+                // mixed: Conversations should never be created with mixed protocol, that's why we break here
+                return ZMConversation?.none
+            }
+            return conversation
+        }) else {
+            return
+        }
+
+        // If this is an mls conversation, then the initial participants won't have
+        // been added yet on the backend. This means that we must take the list of
+        // participants from the action instead of the local conversation.
```

Comment:
Perhaps this comment isn't needed anymore.

Reply:
removed